breadbox/breadbox/schemas/dataset.py

```diff
@@ -192,6 +178,27 @@ def check_valid_value_type_with_allowed_values(self):
             )
         return self
 
+    @field_validator("allowed_values", mode="after")
+    @classmethod
+    def check_valid_allowed_values(cls, v: Optional[List[str]]):
+        """
+        Checks there are no empty strings and no repeated allowed values. Values in allowed values list are not case-sensitive.
+        """
+        if v is None:
+            return v
+        # Decision to make allowed values not case-sensitive in case user error in accidental repeats
+        allowed_values_list_lower = [str(x).lower() for x in v]
+        allowed_values_set = set(allowed_values_list_lower)
+        print(allowed_values_set, allowed_values_list_lower)
+        if len(allowed_values_set) != len(v):
+            raise UserError(
+                msg="Make sure there are no repeats in allowed_values. Values are not considered case-sensitive",
+            )
+        for val in allowed_values_set:
+            if val == "":
+                raise UserError("Empty strings are not allowed!")
```

Comment:
Reading this error, I worry that it doesn't provide enough context. If I saw it when doing an upload, I might not realize that this was an error related to allowed_values.

How about the following?
```
raise UserError("Empty strings are not allowed in allowed_values!")
```

Reply:
Good point!